### ISSUE TITLE
SC2086: Shellcheck Mangles Closing Double Quote Position, Producing Syntax Error in Suggested Edit

### ISSUE THREAD

**Issue by dguyton** [koalaman/shellcheck#1453]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.6.0
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

Could not repro it on Web version of Shellcheck. However, I did not paste my entire script.

Shellcheck provides inconsistent and sometimes grossly incorrect edit suggestions, at times recommending mincing the THEN portion of grammatically correct IF/THEN statements. At times it suggests placing a closing double quote in an obviously incorrect location. I cannot determine a pattern when it creates these erroneous messages. 

See examples below from a BASH script where Shellcheck is complaining incorrectly. This was run on an Ubuntu 16.04.5 LTS server with Shellcheck v0.6.0. Notice the inconsistent advice from grammatically identical expressions (only the var name changes).

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/your/interpreter
your script here

'if [ "$(ls -A $noart_dir)" ]; then # 2/`

```` 

#### [Notice where it recommends positioning the closing double quote.]

In mp3-sorter-v36.sh line 3378:
if [ "$(ls -A $working_dir)" ]; then # 1/ anything in there?
              ^----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
if [ "$(ls -A "$working_dir")" ]; then # 1/ anything in there?

In mp3-sorter-v36.sh line 3380:
        if [ "$(ls -A $working_dir)" ]; then # 2/ sum ting wong
                      ^----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        if [ "$(ls -A $workin"g_dir)" ]; t"hen # 2/ sum ting wong

In mp3-sorter-v36.sh line 3394:
if [ "$(ls -A $unknown_dir)" ]; then # 1/
              ^----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
if [ "$(ls -A "$unknown_dir")" ]; then # 1/

#### Below (line 3403) is the only response that makes sense and can be viewed as correct syntax, though I believe the original is not 'wrong' per se either):

In mp3-sorter-v36.sh line 3403:
if [ "$(ls -A $dupe_dir)" ]; then # 1/
              ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
if [ "$(ls -A "$dupe_dir")" ]; then # 1/

In mp3-sorter-v36.sh line 3413:
        if [ "$(ls -A $noart_dir)" ]; then # 2/
                      ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        if [ "$(ls -A $noart_"dir)" ]; t"hen # 2/   #### < < < Look where it wants me to place the closing "

#### Here's what I wanted or expected to see:

Nothing (I believe my code is grammatically correct).

**Comment by matthewpersico** [comment]
The double quote messages are correct. The correction suggestions are really weird.

**Comment by koalaman** [comment]
This looks like #1420.

You appear to be using the latest git release, which may sometimes be a bit weird. Sadly, there's not really a great way to tell whether you have the v0.6.0 stable release (which didn't have edit suggestions), or a git release which still says v0.6.0 because the version number hasn't been updated.

How are you installing/running shellcheck? Can you try with a stable version, or alternatively with the latest release?

**Comment by dguyton** [comment]
Hey @koalaman. Thank you for building and maintaining shellcheck! It has saved me hours of manual debugging!

I'm using Ubuntu 16.04.5 server; I pulled the latest stable Debian build from git (master branch) a few days after it dropped in early December. I noticed the version I had installed via apt was quite old. I've since modified the BASH code which produced the anomalous response from shellcheck. 

I'll need to find an archived script and setup a test env to try a different build - presumably the Jan 22 2019 version I see currently in the bucket. I'll report back here after I re-test.

**Comment by dguyton** [comment]
Apologies I never followed-up on this. The problem has since resolved itself. Chalking this up to an anomaly with an old build.
